core-redirects-cae/src/main/java/com/tallence/core/redirects/cae/service/SiteRedirects.java

```diff
@@ -39,9 +41,9 @@
   private static final Logger LOG = LoggerFactory.getLogger(SiteRedirects.class);
 
   private String siteId;
-  private final ConcurrentHashMap<String, Redirect> plainRedirects = new ConcurrentHashMap<>();
+  private final ConcurrentHashMap<String, List<Redirect>> plainRedirects = new ConcurrentHashMap<>();
```

Comment:
Und schwupps wird aus O(1) ein O(n)...
Wobei O(n) ja eher die Ausnahme ist. In den meisten Fällen werden die Listen ja Länge von 1 sein..

Reply:
Ja, sehe ich auch so. Das sollte kein Performance-Problem darstellen. Es sei denn, jemand legt für eine Source-Url > 10k Redirects mit unterschiedlichen Url-Parametern an. Aber das fällt m.E. unter "grobe Bedienfehler"